internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -796,6 +792,29 @@ func (r *ClusterDeploymentsReconciler) updateStatus(ctx context.Context, cluster
 	return ctrl.Result{}, nil
 }
 
+func (r *ClusterDeploymentsReconciler) getNumOfClusterAgents(ctx context.Context, clusterInstall *hiveext.AgentClusterInstall, c *common.Cluster) (int, int, error) {
+	if ready, _ := r.ClusterApi.IsReadyForInstallation(c); !ready {
```

Comment:
I don't think that this call is related to `getNumOfClusterAgents`, besides you need to check for errors.